internal/provider/resource_ko_build.go

```diff
@@ -18,105 +17,239 @@ import (
+		r.repo = os.Getenv("KO_DOCKER_REPO")
+	}
+
+	if popts.auth != nil {
+		r.keychain = authn.NewMultiKeychain(staticKeychain{
+			repo: r.repo,
+			b:    popts.auth,
+		}, keychain)
+	} else {
+		r.keychain = keychain
+	}
 }
 
-func resourceBuild() *schema.Resource {
-	return &schema.Resource{
-		// This description is used by the documentation generator and the language server.
-		Description: "Sample resource in the Terraform provider scaffolding.",
-
-		CreateContext: resourceKoBuildCreate,
-		ReadContext:   resourceKoBuildRead,
-		DeleteContext: resourceKoBuildDelete,
-
-		SchemaVersion: 1,
-
-		Schema: map[string]*schema.Schema{
-			ImportPathKey: {
-				Description: "import path to build",
-				Type:        schema.TypeString,
-				Required:    true,
-				ValidateDiagFunc: func(data interface{}, path cty.Path) diag.Diagnostics {
-					// TODO: validate stuff here.
-					return nil
-				},
-				ForceNew: true, // Any time this changes, don't try to update in-place, just create it.
+func (r *BuildResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_build"
+}
+
+func (r *BuildResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"importpath": schema.StringAttribute{
+				Description:   "import path to build",
+				Required:      true,
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+			"working_dir": schema.StringAttribute{
+				Description:   "working directory to build from",
+				Optional:      true,
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
 			},
-			WorkingDirKey: {
-				Description: "working directory for the build",
-				Optional:    true,
-				Default:     ".",
-				Type:        schema.TypeString,
-				ForceNew:    true, // Any time this changes, don't try to update in-place, just create it.
+			"platforms": schema.ListAttribute{
+				Description:   "platforms to build for",
+				Optional:      true,
+				Computed:      true,
+				ElementType:   basetypes.StringType{},
+				Default:       listdefault.StaticValue(types.ListValueMust(basetypes.StringType{}, []attr.Value{basetypes.NewStringValue("linux/amd64")})),
+				PlanModifiers: []planmodifier.List{listplanmodifier.RequiresReplace()},
+				// TODO: validate platforms here.
 			},
-			PlatformsKey: {
-				Description: "Which platform to use when pulling a multi-platform base. Format: all | <os>[/<arch>[/<variant>]][,platform]*",
-				Optional:    true,
-				Type:        schema.TypeList,
-				Elem:        &schema.Schema{Type: schema.TypeString},
-				ForceNew:    true, // Any time this changes, don't try to update in-place, just create it.
+			"base_image": schema.StringAttribute{
+				Description:   "base image to use",
+				Optional:      true,
+				Computed:      true,
```

Comment:
Computed? 🤔

Reply:
When a value has a default, TF makes you mark it as `computed`. Otherwise:

```
        Error: failed to read provider configuration schema for registry.terraform.io/hashicorp/ko: failed to retrieve schema from provider "registry.terraform.io/hashicorp/ko": Schema Using Attribute Default For Non-Computed Attribute: Attribute "base_image" must be computed when using default. This is an issue with the provider and should be reported to the provider developers.
```